### PR TITLE
Step 2

### PR DESCRIPTION
**Pull request by MalpenZibo**

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -1,37 +1,75 @@
...
+        on_success:
+          <<: *up-settings
+          params:
+            <<: *default-up-params
+            status: success
+        on_failure:
+          <<: *up-settings
+          params:
+            <<: *default-up-params
+            status: failure
```
Comment (mcastellini) [inline ci/pipeline.yml:75]: These on_* are on the wrong level and they lack a step where they deal with the job erroring.

File: ci/pipeline.yml [hunk 2]
```diff
@@ -1,37 +1,75 @@
...
     source:
       branch: master
       uri: git@github.com:MalpenZibo/rps.git
       private_key: ((repo-key))
+  - name: pr-branch
+    type: pull-request
+    webhook_token: V21T8e3dLyFWT434NwjL
+    source:
+      repository: MalpenZibo/rps
+      access_token: ((github-token))
```
Comment (mcastellini) [inline ci/pipeline.yml:21]: Let's adjust the polling interval if we are using a webhook token

File: ci/pipeline.yml [hunk 3]
```diff
@@ -1,37 +1,75 @@
...
+        trigger: true
+        version: every
+      - &up-settings
+        put: update-status
+        resource: pr-branch
+        params: &default-up-params
+          path: root
+          status: pending
+      - task: sbt-test
+        config: *test-config
```
Comment (mcastellini) [inline ci/pipeline.yml:65]: Not 100% sure this will interpolate correctly (but I may be wrong)

**Comment by MalpenZibo** [comment]
> Some changes needed. Also, and is my fault for not having explicitly said it, please complete all the `misc` steps before submitting a PR.

The terraform part because I had created the webhooks manually and thought it was fine

File: ci/pipeline.yml (hunk 1 above)
Comment (MalpenZibo) [reply ci/pipeline.yml:75]: uh? ok for the erroring but the other seems to work fine https://concourse.our.buildo.io/teams/buildo/pipelines/simone-rps/jobs/pr-test/builds/1

File: ci/pipeline.yml (hunk 3 above)
Comment (MalpenZibo) [reply ci/pipeline.yml:65]: it seems to work

File: ci/pipeline.yml (hunk 2 above)
Comment (MalpenZibo) [reply ci/pipeline.yml:21]: ? I have used a configuration similar to what is used for gdsm

File: ci/pipeline.yml (hunk 1 above)
Comment (mcastellini) [reply ci/pipeline.yml:75]: It's not that they don't cause an error, it is due to the fact that they are defined at task level, not at job level. If another task or get resource were to fail the corrisponding on_ would not be triggered

Comment (mcastellini) [reply ci/pipeline.yml:75]: More info here: https://concourse-ci.org/hooks-example.html

File: ci/pipeline.yml (hunk 2 above)
Comment (mcastellini) [reply ci/pipeline.yml:21]: Specifying a webhook_token does not change the default refresh interval (see: https://concourse-ci.org/resources.html#schema.resource.check_every)

File: ci/pipeline.yml (hunk 3 above)
Comment (mcastellini) [reply ci/pipeline.yml:65]: Then I am wrong 😄

**Comment by mcastellini** [comment]
Different resource types have different `params` sections. The first is for `teliaoss/github-pr-resource` while the second is for `jtarchie/pr`

File: ci/pipeline.yml (hunk 2 above)
Comment (MalpenZibo) [reply ci/pipeline.yml:21]: hum... There's no way to disable refresh interval?

File: ci/pipeline.yml (hunk 3 above)
Comment (mcastellini) [reply ci/pipeline.yml:65]: IIRC using this syntax instead of `<<: *test-config` breaks when merging fields

**Pull request merged**